### PR TITLE
fix RareClusterStateIT

### PR DESCRIPTION
**Pull request by idegtiarenko**

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java [hunk 1]
```diff
@@ -140,7 +141,12 @@ private <Req extends ActionRequest, Res extends ActionResponse> ActionFuture<Res
             }
         });
         ActionFuture<Res> future = req.execute();
-        assertBusy(() -> assertTrue(masterCoordinator.cancelCommittedPublication()));
+        // cancel all commit publications produced by the request
+        internalCluster().getCurrentMasterNodeInstance(ThreadPool.class).generic().execute(() -> {
+            while (!future.isDone()) {
```
Comment (DaveCTurner) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java:146]: I'm not sure that's enough - IIRC deleting an index may trigger another reroute but it doesn't wait for that reroute to complete.

**Review by DaveCTurner — COMMENTED** [review]
Looks good, but should we make this change in `master` first?

**Comment by idegtiarenko** [comment]
>Looks good, but should we make this change in master first?

Okay, let me change the base for this pr

**Comment by idegtiarenko** [comment]
Done, this is now targeting master

**Review by DaveCTurner — APPROVED** [review]
Nice, LGTM

**Pull request merged**